crates/extensions/c8y_http_proxy/src/actor.rs

```diff
@@ -186,56 +197,156 @@ impl C8YHttpProxyActor {
         &mut self,
         device_id: Option<&str>,
     ) -> Result<String, C8YRestError> {
-        let url_get_id = self.end_point.get_url_for_get_id(device_id);
+        let internal_id = if self.cached_identifiers.main_device_internal_id.is_none() {
```

Comment:
There is a weird overlapping of responsibilities between the `try_get_and_set_internal_id` and `try_get_internal_id`. Both are manipulating the cache leading to useless and hard to follow sequences (as storing in the cache what has just been retrieved form the cache.

- The `try_get_internal_id` should not even be aware that a cache exist.
- The `try_get_and_set_internal_id` should call the former only when the id is not cached.

Reply:
Refactored